pkg/receiver/jobreceiver/runner.go

```diff
@@ -13,52 +19,121 @@ import (
 // Build returns the job runner, the process responsible for scheduling and
 // running commands and piping their output to the output consumer.
 func (c Config) Build(logger *zap.SugaredLogger, out consumer.Interface) (builder.JobRunner, error) {
-	return &stubRunner{
-		Exec:     c.Exec,
-		Consumer: out,
+	return &runner{
+		exec:     c.Exec,
+		schedule: c.Schedule,
+		consumer: out,
+		logger:   logger,
 	}, nil
 }
 
-// stubRunner is a stub implementation.
-type stubRunner struct {
-	Exec     ExecutionConfig
-	Consumer consumer.Interface
+// runner schedules and executes commands
+type runner struct {
+	exec     ExecutionConfig
+	schedule ScheduleConfig
+
+	consumer consumer.Interface
+
+	logger *zap.SugaredLogger
+
+	wg     sync.WaitGroup
+	cancel func()
 }
 
-// Start stub impl. runs command once at startup then idles indefinitely.
-func (r *stubRunner) Start(operator.Persister) error {
+// Start running the configured command.
+//
+// Returns after starting a goroutine that will:
+// 1. Ensures all runtime assets are downloaded and available
+// 2. Build the command environment the monitoring job will be executed with
+// 3. Begin scheduling the monitoring job
+func (r *runner) Start(operator.Persister) error {
+
+	ctx := context.WithValue(context.Background(), consumer.ContextKeyCommandName, r.exec.Command)
+	ctx, r.cancel = context.WithCancel(ctx)
+
+	//TODO expose asset fetching http, retry and storage strategy as user
+	//configuration?
+	assetManager := asset.Manager{
+		Fetcher: asset.NewFetcher(
+			r.logger,
+			&http.Client{
+				Transport: http.DefaultTransport.(*http.Transport).Clone(),
+			},
+		),
+		StoragePath: os.TempDir(),
+		Logger:      r.logger,
+	}
+
+	runtimeAssets := r.exec.RuntimeAssets
+	if err := assetManager.Validate(runtimeAssets); err != nil {
+		return fmt.Errorf("invalid runtime asset(s): %w", err)
+	}
+
+	r.wg.Add(1)
 	go func() {
-		ctx := context.Background()
+		defer r.wg.Done()
+		refs, err := assetManager.InstallAll(ctx, runtimeAssets)
+		if err != nil {
+			r.logger.Errorf("failed to load runtime assets. monitoringjob will not be scheduled: %s", err)
+		}
+		commandEnv := os.Environ()
+		for _, ref := range refs {
+			commandEnv = ref.MergeEnvironment(commandEnv)
```

Comment:
return an error if the environment variable validation fails.

Reply:
Per above comment, there is no validation to be done. We're simply merging os.Environ's PATH (and a few other vars) with some extra entries.

i.e. `PATH=/path/to/asset1/bin:/path/to/asset2/bin:$PATH`